src/term-spec.js

```diff
@@ -206,19 +330,39 @@ declare export class ExportLocalSpecifier extends Term {
 
 // property definition
 declare export class Method extends MethodDefinition {
+  parent?: ClassExpression | ClassDeclaration | ObjectExpression;
   isAsync: any;
   isGenerator: any;
   params: FormalParameters;
 }
 
-declare export class Getter extends MethodDefinition { }
+declare export class AbstractMethod extends AbstractMethodDefinition {
+  parent?: ClassExpression | ClassDeclaration;
+  isAsync: any; // boolean
+  isGenerator: any; // boolean
+  params: FormalParameters;
+}
+
+declare export class Getter extends MethodDefinition {
+  parent?: ClassExpression | ClassDeclaration | ObjectExpression;
+}
+
+declare export class AbstractGetter extends AbstractMethodDefinition {
+  parent?: ClassExpression | ClassDeclaration;
+}
 
 declare export class Setter extends MethodDefinition {
-  param: ObjectBinding | ArrayBinding | BindingIdentifier | MemberExpression | BindingWithDefault;
-  // param: Binding or BindingWithDefault;
+  parent?: ClassExpression | ClassDeclaration | ObjectExpression;
+  param: ParameterDeclaration;
+}
+
+declare export class AbstractSetter extends AbstractMethodDefinition {
+  parent?: ClassExpression | ClassDeclaration;
+  param: ParameterDeclaration;
 }
 
 declare export class DataProperty extends NamedObjectProperty {
+  hasQuestionToken: any; // boolean
```

Comment:
I think I may have overdone it with the question tokens. I think I added them only where the TypeScript AST permits them but some of them (like this one, but also the one on `FunctionDeclaration`) don't seem to make too much sense...